src/H5FDsubfiling/H5FDioc.c

```diff
@@ -841,6 +843,12 @@ H5FD__ioc_open(const char *name, unsigned flags, hid_t fapl_id, haddr_t maxaddr)
     /* Fill in the file config values */
     HDmemcpy(&file_ptr->fa, config_ptr, sizeof(H5FD_ioc_config_t));
 
+    /* Copy the ioc FAPL. */
+    if (H5FD__copy_plist(config_ptr->under_fapl_id, &(file_ptr->fa.under_fapl_id)) < 0) {
+        file_ptr->fa.under_fapl_id = H5I_INVALID_HID;
```

Comment:
The tools code tries all available VFDs when it can't open with sec2. Here it tried Subfiling and when Subfiling couldn't resolve file paths below (when it didn't create the files), it was closing the file FAPL out from under the tools code and causing an infinite loop.